lib/takeaway.rb

```diff
@@ -0,0 +1,78 @@
+#require 'rubygems'
+require 'twilio-ruby'
+
+class TakeAway
+
+  attr_reader :menu
+  attr_accessor :order, :prices
+
+  def initialize
+    @order = {}
+    @prices = []
+  end
+
+  def add_menu(menu)
+    @menu = menu.dishes
+  end
+
+  def show_menu
+    menu
+  end
+
+  def place_order(items={})
+    items.each do |dish, quantity|
+      raise "#{dish.capitalize} is not on the menu." unless menu.include?(dish)
+      order["#{dish} x #{quantity}"] = total_cost_per_item(menu[dish], quantity)
+    end
+  end
+
+
+  def grand_total
+    total_in_pence = prices.inject(:+)
+    "%.2f" % (total_in_pence / 100.0)
+  end
+
+  def total_cost_per_item(price, quantity)
+    price_in_pence = ((price * 100).round * quantity)
+    prices << price_in_pence
+    "%.2f" % (price_in_pence / 100.0)
+  end
+
+  def show_order
+    order_output = print_order_items
+    order_output += "Total: £#{grand_total}"
+    order_output
+  end
+
+  def delete_from_order(dish, quantity)
+    order.delete("#{dish} x #{quantity}")
+  end
+
+  def confirm_order
+    puts "Please confirm that your order is correct by typing Y for Yes or N for No."
```

Comment:
Line is too long. [85/80]<br>Do not write to stdout. Use Rails' logger if you want to log.